### PR TITLE
Feat: add gc_content() calculation method to fgpyo/sequence.py

### PR DESCRIPTION
**Pull request by emmcauley**
Add method to calculate fraction GC content in a string of bases. Corresponding unit tests also added.

### CODE REVIEW

File: fgpyo/sequence.py [hunk 1]
```diff
@@ -70,3 +70,12 @@ def reverse_complement(bases: str) -> str:
...
     return "".join([_COMPLEMENTS[b] for b in bases[::-1]])
+
+
+def gc_content(bases: str) -> float:
+    """Calculates the fraction of G and C bases in a sequence."""
+    if bases:
+        gc_count = sum(1 if (ch == "C" or ch == "G") else 0 for ch in bases)
+        return gc_count / len(bases)
+    else:
+        return 0
```
Comment (nh13) [inline fgpyo/sequence.py:81]: 1. Reminder to use `if len(str) == 0:` instead of the implicit conversion `if str:`.   This is included in my suggestion below.
2. Do we handle lower-case bases?  This is not included in my suggestion below.
```suggestion
    if len(bases) = 0:
        return 0
    gc_count = sum(1 if (ch == "C" or ch == "G") else 0 for ch in bases)
    return gc_count / len(bases)
```

File: fgpyo/tests/test_sequence.py [hunk 2]
```diff
@@ -15,3 +16,11 @@ def test_reverse_complement() -> None:
...
     with pytest.raises(KeyError):
         reverse_complement("ABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+
+def test_gc_content() -> None:
+    assert gc_content("ATATATATTATATA") == 0.0
+    assert gc_content("GCGCGCGCGCGCG") == 1.0
+    assert gc_content("ACGT") == 0.5
+    assert gc_content("") == 0.0
+    assert gc_content("ACGTN") == 0.4
```
Comment (nh13) [inline fgpyo/tests/test_sequence.py:26]: 1. Can you make these parameterized with pytest?
2. After (1), then can you test both upper and lower case (you can just have two asserts in the test method, one that upper-cases explicitly, and the other that lower-cases explicitly)

File: docs/conf.py [hunk 3]
```diff
@@ -10,7 +10,8 @@
 # All configuration values have a default; values that are commented out
 # serve to show the default.
 
-import sys, os, glob
+import os
```
Comment (nh13) [inline docs/conf.py:13]: where are these changes coming from?  I think these are unrelated.  Can you either remove, or make a separate PR?

**Review by nh13 — CHANGES_REQUESTED** [review]

File: docs/conf.py (hunk 3 above)
Comment (emmcauley) [reply docs/conf.py]: I think they resulted from a `ruff format` run; it should be fixed now

File: fgpyo/tests/test_sequence.py (hunk 2 above)
Comment (emmcauley) [reply fgpyo/tests/test_sequence.py]: 1. parameterized with pytest
2. unit test includes mixed base cases

File: fgpyo/sequence.py (hunk 1 above)
Comment (emmcauley) [reply fgpyo/sequence.py]: thank you! Suggestion implemented and I use .upper() on the input sequence, so now we should be handling lower and upper cases appropriately

File: fgpyo/sequence.py [hunk 4]
```diff
@@ -70,3 +70,12 @@ def reverse_complement(bases: str) -> str:
...
     return "".join([_COMPLEMENTS[b] for b in bases[::-1]])
+
+
+def gc_content(bases: str) -> float:
+    """Calculates the fraction of G and C bases in a sequence."""
+    if len(bases) == 0:
+        return 0
+    else:
+        gc_count = sum(base == "C" or base == "G" for base in bases.upper())
+        return gc_count / len(bases)
```
Comment (nh13) [inline fgpyo/sequence.py:81]:
```suggestion
    gc_count = sum(1 for base in bases if base == "C" or base == "G" or base == "c" or base == "g")
    return gc_count / len(bases)
```

**Review by nh13 — CHANGES_REQUESTED** [review]
Minor suggestion that avoids having to traverse `bases` twice (once to uppercase, once to compare).  Also avoids relying summing an iterator of bools, which seems nonsensical (bool is not a numeric), but that's python

**Review by nh13 — APPROVED** [review]
LGTM, merge when ready:
punch it
